client/network/sync/src/lib.rs

```diff
@@ -1725,6 +1732,27 @@ where
 
 		Ok(OpaqueStateResponse(Box::new(response)))
 	}
+
+	fn poll(
+		&mut self,
+		cx: &mut std::task::Context,
+	) -> Poll<PollBlockAnnounceValidation<B::Header>> {
+		while let Poll::Ready(Some(event)) = self.service_rx.poll_next_unpin(cx) {
+			match event {
+				ToServiceCommand::SetSyncForkRequest(peers, hash, number) => {
+					self.set_sync_fork_request(peers, &hash, number);
+				},
+			}
+		}
+
+		match self.block_announce_validation.poll_next_unpin(cx) {
+			Poll::Ready(Some(res)) => {
+				self.peer_block_announce_validation_finished(&res);
+				Poll::Ready(self.finish_block_announce_validation(res))
+			},
+			_ => Poll::Pending,
+		}
+	}
```

Comment:
This `poll()` should be called the same way `poll_block_announce_validation()` was called so it must be polled until it returns `Poll::Pending` so that all block announce validation messages are consumed. I've modified the code to call `poll_block_announce_validation()` as it should have all along.

Once the `ChainSync` can handle block announce validations without the help of `Protocol`, the latter `poll_next_unpin()` can also be converted into a loop.